### PR TITLE
Improving routing and navigation

### LINKED ISSUE #29: Fix routing in build
Once built and deployed, routing doesn't seem to be handled

We had to change some nextjs settings for the data dashboard to build in a way that works for s3, probably need to do the same thing for vite

### PR DESCRIPTION
**Pull request by devinmatte**
## Motivation

Get navigation and routing working better, as well as a few more UI cleanups

## Changes

- Fixes loading and navigation between shutdowns and line colors
- Swaps `/home` to `/` to avoid redundant routing names
- Fixes labels on charts to be slightly clearer
Screenshot 2024-04-22 at 7 04 00 PM

- Swaps chart color so primary is for after shutdown
- Ensures routing works in s3 with cloudformation (fixes #29)
- Slightly improving types

## Testing Instructions

Check out on the url

### CODE REVIEW

File: src/components/LineGraph.tsx [hunk 1]
```diff
@@ -149,10 +165,17 @@ const LineGraph = () => {
...
-                const { id, x } = elements[0].element['$context'].raw;
-                scroller.scrollTo(`${id}-${x[0]}-${x[1]}`, {
-                  smooth: true,
-                  offset: -32,
+                const {
+                  x: [startDate, endDate],
+                  start_station,
+                  end_station,
+                  line,
+                } = elements[0].element['$context'].raw;
```
Comment (skaplan-dev) [inline src/components/LineGraph.tsx:173]: you'd think there would be an easier way to access this lol

File: src/types.ts [hunk 2]
```diff
@@ -31,8 +31,8 @@ export interface LineMap {
 export type Shutdown = {
   start_date: string;
   stop_date: string;
-  start_station: Station;
-  end_station: Station;
+  start_station: Station | undefined;
```
Comment (skaplan-dev) [inline src/types.ts:34]: could we do `start_station?: Station` etc

File: src/utils/stations.ts [hunk 3]
```diff
@@ -46,7 +46,11 @@ export const getStationsFromShutdown = (stop: string, line: Lines) => {
 };
 
 // This assumes the stops are on the same line
-export const findStopsBetween = (startStation: string, endStation: string, line: Lines) => {
+export const findStopsBetween = (
+  startStation: string | undefined,
+  endStation: string | undefined,
```
Comment (skaplan-dev) [inline src/utils/stations.ts:51]: we can just do `?` here in the type

**Review by skaplan-dev — APPROVED** [review]

File: src/utils/stations.ts (hunk 3 above)
Comment (devinmatte) [reply src/utils/stations.ts:51]: So the advantage of being explicit is that it still requires you pass something into the prop, even if it's undefined, but you can't just forget to include the prop

Comment (skaplan-dev) [reply src/utils/stations.ts:51]: ahh gotcha

**Pull request merged**